File: ruby/problem_6.rb
#!/bin/env ruby
#
# Problem 6
#
# Author::    mattt
#
require 'ruby/euler_lib'

timed_run{
  sum_of_squares = 0

  sum = 0
  (1..100).each do |num|
    sum_of_squares += num ** 2
    sum += num
  end

  square_of_sums = sum ** 2

  #25164150
  puts square_of_sums - sum_of_squares
}

File: golang/problem_6.go
package main

func problem6() int {
	sumOfSquares := 0

	sum := 0
	for num := 1; num <= 100; num++ {
		sumOfSquares += num * num
		sum += num
	}

	squareOfSums := sum * sum

	return squareOfSums - sumOfSquares
}
